From 8e71c608126c2acf4bc9806622f3bf67e341913f Mon Sep 17 00:00:00 2001
From: Kranthi G <gudishax.kranthikumar@intel.com>
Date: Wed, 18 Nov 2015 15:26:17 +0530
Subject: [PATCH 0628/2313] ASoC: Intel: Skylake: Add updown mixer changes

As firmware structure modified for updown mixer,
this commit will add channel map member to updown
mixer structure and also changes the size of coefficints to 8.

Change-Id: Ie825def1852a5d2d81e7bf5b1e5d9b2138362211
Signed-off-by: Kranthi G <gudishax.kranthikumar@intel.com>
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
Signed-off-by: Guneshwor Singh <guneshwor.o.singh@intel.com>
---
 sound/soc/intel/skylake/skl-messages.c | 3 ++-
 sound/soc/intel/skylake/skl-topology.h | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index 02e247f..36ff1a6 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -724,10 +724,11 @@ static void skl_set_updown_mixer_format(struct skl_sst *ctx,
 
 	/* Select F/W default coefficient */
 	mixer_mconfig->coeff_sel = 0x0;
+	mixer_mconfig->ch_map = fmt->ch_map;
 
 	/* User coeff, don't care since we are selecting F/W defaults */
 	for (i = 0; i < UP_DOWN_MIXER_MAX_COEFF; i++)
-		mixer_mconfig->coeff[i] = 0xDEADBEEF;
+		mixer_mconfig->coeff[i] = 0x0;
 }
 
 /*
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index fdf0f69..fc3e8c4 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -34,7 +34,7 @@
 #define MAX_FIXED_DMIC_PARAMS_SIZE 727
 
 /* Maximum number of coefficients up down mixer module */
-#define UP_DOWN_MIXER_MAX_COEFF		6
+#define UP_DOWN_MIXER_MAX_COEFF		8
 
 #define MODULE_MAX_IN_PINS	8
 #define MODULE_MAX_OUT_PINS	8
@@ -161,6 +161,7 @@ struct skl_up_down_mixer_cfg {
 	u32 coeff_sel;
 	/* Pass the user coeff in this array */
 	s32 coeff[UP_DOWN_MIXER_MAX_COEFF];
+	u32 ch_map;
 } __packed;
 
 struct skl_algo_cfg {
-- 
2.7.4

